proposals/TestProposals.sol

```diff
@@ -41,20 +41,34 @@ contract TestProposals is Test {
     bool public doTeardown;
     bool public doValidate;
 
-    address public deployer = address(0x00000108);
+    address public deployer;
 
     function setUp() public {
-        addresses = new Addresses();
-
-        // Load proposals
-        proposals.push(Proposal(address(new zip000()))); /// Genesis token proposal
-        proposals.push(Proposal(address(new zip001()))); /// Wearables, Core, ADMIN_MULTISIG proposal
-        proposals.push(Proposal(address(new zip002()))); /// Timelock proposal
-        proposals.push(Proposal(address(new zip003()))); /// CGv1 proposal
-        proposals.push(Proposal(address(new zip004()))); /// TokenIds, MaxSupply and Capsule settings proposal
-        proposals.push(Proposal(address(new zip005()))); /// MaxSupply settings proposal
-        proposals.push(Proposal(address(new zip006()))); /// MaxSupply settings proposal
-        proposals.push(Proposal(address(new zip007()))); /// MaxSupply settings proposal
+        string memory environment = vm.envOr("ENVIRONMENT", string("localnet"));
+        string memory addressPath = string(abi.encodePacked("proposals/Addresses/", environment, ".json"));
+        addresses = new Addresses(addressPath);
+
+        bool bootstrap = vm.envOr("BOOTSTRAP", false);
+
+        // Admin access for deployer(ZTX_DEPLOYER) is revoked on mainnet
+        if (block.chainid == 42161) {
+            deployer = addresses.getAddress("ADMIN_MULTISIG");
+        } else {
+            deployer = addresses.getAddress("DEPLOYER");
+        }
```

Comment:
Perhaps we could update this in another PR, but am thinking that all operations should generate calldata, even the first proposal.

Reply:
I also don't think we should be checking for an explicit chain Id.

Reply:
Earlier the integration tests were not running against the deployed contracts. We were redeploying all the contracts and updating the addresses, so in a way we were not doing integration testing. When I ran integration tests against mainnet contracts I got to understand that we revoke the admin role from the deployer on mainnet which means for testing against mainnet we have to make the deployer as `ADMIN_MULTISIG`.
Please let me know if I am missing something here.

Reply:
I just don't like this. We're now hardcoding a chain ID.

Reply:
What if we have a DEPLOYER address in the address file? In `mainnet.json` it can be the `ADMIN_MULTISIG` address, and in `devnet.json` (for example) it doesn't need to change. Make sense? This way we don't need this if statement. It'll just use what's in the json file.

Reply:
I thought of doing this but  this way we will be naming `ADMIN_MULTISIG` as `DEPLOYER` in json which is not the case. I think the current code explains the real scenario more clearly. What do u think.